python/ray/experimental/state/state_cli.py

```diff
@@ -669,3 +677,321 @@ def object_summary(ctx, timeout: float, address: str):
             ),
         )
     )
+
+
+@click.group("logs", invoke_without_command=True)
+@click.pass_context
+def logs_state_cli_group(ctx):
+    """Get logs based on filename (file) or resource identifiers (actor)
+
+    Usage:
+
+        Get all the log files available on a node (ray address could be
+        obtained from `ray start --head` or `ray.init()`)
+
+        ```
+        ray log --address=<ray address>
+        ```
+
+        [ray logs file] Print the last 500 lines of raylet.out on a head node.
+
+        ```
+        ray logs file raylet.out --tail 500
+        ```
+
+        [ray logs actor] Follow the log file with an actor id ABC.
+
+        ```
+        ray logs actor --id ABC --follow
+        ```
+    """
+    if ctx.invoked_subcommand is None:
+        # Forward to `ray logs file`
+        ctx.invoke(log_file)
+
+
+def _parse_log_node_target(
+    address: Optional[str], node_id: Optional[str], node_ip: Optional[str]
+):
+    """Parse the query API server and the target nodes from which logs should be retrieved
+
+    Args:
+        address: Address of the GCS of the head node running the API server. This is
+            usually set through RAY_ADDRESS, `auto`, or an explicit <ip:port>.
+        node_id: Id of the node containing the logs.
+        node_ip: Ip of the node containing the logs.
```

Comment:
Returns:
    Tuple of API server URL, node id, and node ip. If no node id and node ip is provided, it will return the head node ip